CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -87,8 +87,45 @@ public class Child {
   /// This is used to e.g. describe the child if all of its tokens are missing in the source file.
   public let nameForDiagnostics: String?
 
-  /// A doc comment describing the child.
-  public let documentation: SwiftSyntax.Trivia
+  /// A summary of a doc comment describing the child. Full docc comment for the
+  /// child is available in ``Child/documentation``, and includes detailed list
+  /// of possible choices for the child if it's a token kind.
+  public let documentationSummary: SwiftSyntax.Trivia
+
+  /// A docc comment describing the child, including the trivia provided when
+  /// initializing the ``Child``, and the list of possible token choices inferred automatically.
+  public var documentation: SwiftSyntax.Trivia {
+    if case .token(let choices, _, _) = kind {
+      let grammar = GrammarGenerator()
+
+      var tokensDocumentation = """
+        ### Tokens
+        For syntax trees generated by the parser, this is guaranteed to be the following
+        """
+
+      tokensDocumentation +=
+        choices.count == 1
+        ? "kind: \(grammar.grammar(for: choices.first!))"
+        : """
+        kinds:
+        \(choices.map { " - \(grammar.grammar(for: $0))" }.joined(separator: "\n"))
+        """
+
+      // trivia's last piece does not end in \n.
+      // So if the existing documentation is not emtpy, we need to append an
+      // additional TriviaPiece.newlines(1) before appending the token choices.
+      if documentationSummary.isEmpty {
```

Comment:
Might be a good idea to provide `Trivia.appending(trivia, newLines: Int = 0)` that inserts new lines between trivia pieces automatically if we see this pattern frequently, but I wanted to keep this change as small as possible.